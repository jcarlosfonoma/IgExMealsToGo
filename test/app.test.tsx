import React from "react"
import { create } from "react-test-renderer"
import App from "../app/app"
import { RestaurantsContextProvider } from "../app/services/restaurants/restaurants.context"
import { FavouritesContextProvider } from "../app/services/favourites/favourites.context"
import { AuthenticationContextProvider } from "../app/services/authentication/authentication.context"
import { LocationContextProvider } from "../app/services/location/location.context"
import { ThemeProvider } from "styled-components/native"
import { theme } from "../app/infrastructure/theme"

const tree = create(
  <ThemeProvider theme={theme}>
    <AuthenticationContextProvider>
      <FavouritesContextProvider>
        <LocationContextProvider>
          <RestaurantsContextProvider>
            <App />
          </RestaurantsContextProvider>
        </LocationContextProvider>
      </FavouritesContextProvider>
    </AuthenticationContextProvider>
  </ThemeProvider>,
)

jest.mock("@expo/vector-icons", () => ({
  Ionicons: "",
}))

jest.mock("@expo-google-fonts/oswald", () => ({
  useFonts: () => [],
  useOswald: () => []
}))

jest.mock("@expo-google-fonts/lato", () => ({
  useFonts: () => [],
  useLato: () => []
}))

jest.mock("firebase", () => {
  const firebaseObject = {
    onAuthStateChanged: () => {
      return null
    },
  }

  return {
    apps: { length: 1 },
    auth: () => {
      return { ...firebaseObject }
    },
  }
})

describe("app test", () => {
  test("app should match snapshot", () => {
    expect(tree).toMatchSnapshot()
  })
})
