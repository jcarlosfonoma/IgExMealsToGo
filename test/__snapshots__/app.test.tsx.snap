// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app test app should match snapshot 1`] = `null`;
